geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/CreateGatewayReceiverCommand.java

```diff
@@ -80,46 +83,71 @@ public Result createGatewayReceiver(@CliOption(key = {CliStrings.GROUP, CliStrin
         executeAndGetFunctionResult(GatewayReceiverCreateFunction.INSTANCE,
             gatewayReceiverFunctionArgs, membersToCreateGatewayReceiverOn);
 
-    CommandResult result = ResultBuilder.buildResult(gatewayReceiverCreateResults);
-
-    XmlEntity xmlEntity = findXmlEntity(gatewayReceiverCreateResults);
-    // no xml needs to be updated, simply return
-    if (xmlEntity == null) {
-      return result;
-    }
+    ResultModel result = ResultModel.createMemberStatusResult(gatewayReceiverCreateResults);
 
     boolean allSuccessful = gatewayReceiverCreateResults.stream()
         .map(CliFunctionResult::isSuccessful).reduce(true, (x, y) -> x && y);
     if (!allSuccessful) {
       result.setStatus(Status.ERROR);
-      return result;
     }
 
-    // has xml but unable to persist to cluster config, need to print warning message and return
-    if (onMember != null || getConfigurationPersistenceService() == null) {
-      result.setCommandPersisted(false);
-      return result;
-    }
-
-    // update cluster config
-    ((InternalConfigurationPersistenceService) getConfigurationPersistenceService())
-        .addXmlEntity(xmlEntity, onGroups);
+    result.setConfigObject(configuration);
     return result;
   }
 
+  @Override
+  public void updateClusterConfig(String group, CacheConfig config, Object configObject) {
+    config.setGatewayReceiver((CacheConfig.GatewayReceiver) configObject);
+  }
+
+  private CacheConfig.GatewayReceiver buildConfiguration(Boolean manualStart, Integer startPort,
+      Integer endPort, String bindAddress, Integer maximumTimeBetweenPings,
+      Integer socketBufferSize, String[] gatewayTransportFilters, String hostnameForSenders) {
+    CacheConfig.GatewayReceiver configuration = new CacheConfig.GatewayReceiver();
+
+    if (gatewayTransportFilters != null) {
+      List<ClassWithParametersType> filters =
+          Arrays.stream(gatewayTransportFilters).map(fullyQualifiedClassName -> {
+            ClassWithParametersType thisFilter = new ClassWithParametersType();
+            thisFilter.setClassName(fullyQualifiedClassName);
+            return thisFilter;
+          }).collect(Collectors.toList());
+      configuration.getGatewayTransportFilter().addAll(filters);
+    }
+    if (startPort != null) {
+      configuration.setStartPort(String.valueOf(startPort));
+    }
+    if (endPort != null) {
+      configuration.setEndPort(String.valueOf(endPort));
+    }
+    configuration.setBindAddress(bindAddress);
+    if (maximumTimeBetweenPings != null) {
+      configuration.setMaximumTimeBetweenPings(String.valueOf(maximumTimeBetweenPings));
+    }
+    if (socketBufferSize != null) {
+      configuration.setSocketBufferSize(String.valueOf(socketBufferSize));
+    }
+    configuration.setHostnameForSenders(hostnameForSenders);
+    configuration.setManualStart(manualStart);
+    return configuration;
+  }
+
   public static class Interceptor extends AbstractCliAroundInterceptor {
```

Comment:
When switching to `ResultModel` you'll need to change any `Interceptor`s too. `preExecution` should return a `ResultModel` and `postExecution` a `CommandResult`. Now would be a good time to add a test for these :). Typically `preExecution` is just validation - a simple approach for testing can be seen in `ExecuteFunctionCommandTest`.

Reply:
That example is still pending review...

Reply:
Oooo, good point.

Looks like the `preExecution` is already checked in the UnitTest, but yeah, good to double check for potential gaps in our coverage.